src/main/scala/tech/sourced/gemini/Gemini.scala

```diff
@@ -55,16 +55,67 @@ class Gemini(session: SparkSession, log: Slf4jLogger, keyspace: String = Gemini.
       .getCommits
       .getTreeEntries
       .getBlobs
+      .filter('is_binary === false)
+
+  def sparkExtractUast(files: DataFrame): DataFrame = {
+    // TODO(max): get languages from bblfsh directly as soon as
+    // https://github.com/bblfsh/client-scala/issues/68 resolved
+    val langs = Array("Java", "Python", "Go", "JavaScript", "TypeScript", "Ruby", "Bash", "Php")
+
+    files
+      .dropDuplicates("blob_id")
+      .classifyLanguages
+      .filter('lang.isin(langs: _*))
+      .extractUASTs
+      .select("repository_id", "path", "blob_id", "uast")
+      .filter(_.getAs[Seq[Array[Byte]]]("uast").nonEmpty)
+      .withColumn("document", expr("CONCAT(repository_id, '//', path, '@', blob_id)"))
+      .select("document", "uast")
+  }
+
+  def sparkFeatures(uasts: DataFrame): RDD[SparkFeature] = {
```

Comment:
Just wondering, if it would be worth adding a few lines of comment here, explaining why DataFrame API \w table-like structure that is used in the rest of the code is not a good fit and RDD of case class is used instead.

Reply:
Mostly because I had problems with implementing it with DataFrame and Apollo use RDD here.
But I agree it's better to use DataFrame. Let me come back to it in sequential PR. For now, I added TODO.